LynnGazis/safeplaces-frontend
# dev_rocket into dev (#133)

* sb setup

* Add sb

* Port nav files from main repo

* render stories

* add crucial styles to all stories

* remove dead code

* adjust paths

* housekeeping

* improvie configuration

* delete welcome story

* add global styles

* Modal component

* records table initial build

* Tweak styles // add new case CTA // move things around

* modal mq

* Table refactor

* Load examples

* uses classes / box-shadows

* tidy

* further tweaks

* Sidebar Actions UI dev

* Sidebar Header UI dev

* Sidebar Wrapper UI dev

* Sidebar stories

* drop it for now

* Button mods for review

* Dialog abstraction

* add modal

* Add New Record modal

* remove redundant story

* code will be provided by API

* use design code

* cherry pie

* Fix button text

* Record Added UI

* Fix button text

* remove

* work buttons

* Leave Session UI

* button font weight adjustment

* button font weight adjustment (really need to get PRs merged)

* cherry picks and globals

* Stage for Pub. UI dev

* remove erroneous prop

* Publish Data UI

* Update header

* Initial development

* initial build

* ad initial hooks (needs review)

* Working slider story

* change unused var lint rule (sorry)

* Separate toggle for cleaner component

* tidy files

* Date Selector initial build

* port wrapper from personal fork (need to sort PRs)

* use actual dates

* Checks

* Include sidebar wrapper for better stories

* Create filter group component

* tweak checkbox config

* add slider styles to global

* Initial duration UI work

* import checkbox tweaks

* tweak slider disabled styles

* Filter group wrapper tweaks

* Duration Filter improvements

* tweaks

* import existing work

* travelling filter UI

* add id to checkbox

* Filter Group improvements

* cherry pick group tweaks

* Selected Data wrapper

* Selected Data Item

* Selected Data progress

* Introduce context menu

* Improve selected data list

* Add click away logic to point contenxt

* Add click away logic to point contenxt

* tidy stories

* add overflow catch to wrapper

* Remove that

* context menu positioning

* scroll logic

* overflow trouble

* Sidebar sadness

* final style tweaks

* // add toast // don't mix managers

* kill it

* temp

* reinstate lock file

* Improve stories / tweak styles

* Add Redactor Tool to story

* tidy stories

* UI fixes

* Rework button

* refactor existing buttons

* Date Selector UI improvements

* Initial actions for the views section

* [PLACES-162] HA first login configuration screen (onboarding)

* initial store code

* create separate map (for now for

* style tweaks and button additions

* [PLACES-142] Login integration, auth duck refactor

* [PLACES-142] Login integration, auth duck refactor + private routes

* solitary record saga

* store legacy components

* global things

* shared components

* Trace view initial

* settings view shell

* Publish initial

* onboarding view

* Auth. view

* Trace child

* removals

* Nav updates

* paths and minor changes

* Map cleaning pt.1

* further path fixes

* Removals

* sidebar header props

* weird issue with login

* build tracer views

* sort tools

* more legacy

* tracer actions

* App tidy

* temp logo route

* temp settings render

* Flesh out publish screen

* clean map file

* fix routing

* remove nav

* add record tweaks

* Add ducks for records and general application state

* fix login form

* delete old images

* fix button type

* progress

* fix login form behaviour

* improve notifications

* revert service / notification styles

* revert store (for now)

* clear status

* render record ID

* record added port

* check for record before rendering

* silly boy

* keep it clean

* don't do that

* logs, man

* Temp. record deletion flow

* [PLACES-237] Dashboard Config BE Integration

* replace record duck for cases duck // update sagas

* update store references // render some location data

* validation and redirect after completing form

* moved input list to its own file, comp was getting huge

* MAP: kill lines, adjust layers

* CASES: improve ducks; service and sagas, fix component refs

* validations

* Maps, man

* fix this, fix that

* remove redundant code

* adjust focus markers

* initial confirm bounds work

* fix modal toggle issue

* toggle ctas and notice based on setting

* validation

* further map cleaning

* onboarding state

* [PLACES-160] LOGIC - Generate new case instance (Mobile Data)

* partially replaced mocked user from login

* tidy logs // add bounds selector

* 🌍😡🆘

* map defaults / saga

* need GPS data first

* [PLACES-170] Settings UI + Pre-populate inputs

* improve record action styles

* hide static elements

* config modal tweaks

* style tweaks

* provide out

* improve header rendering logic

* defaulted day slider to zero

* refine layout

* map housekeeping

* fix stories (file paths)

Co-authored-by: FacundoMalgieri <facundo.malgieri@gmail.com>
